apps/desktop-wallet/src/api/apiDataHooks/useAddressesAlphBalances.ts

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+import { useQueries, UseQueryResult } from '@tanstack/react-query'
+
+import { addressAlphBalancesQuery, AddressAlphBalancesQueryFnData } from '@/api/addressQueries'
+import { useAddressesLastTransactionHashes } from '@/api/addressTransactionsDataHooks'
+import { combineIsLoading } from '@/api/apiDataHooks/utils'
+import { useAppSelector } from '@/hooks/redux'
+import { DisplayBalances } from '@/types/tokens'
+
+// Using undefined to avoid adding noUncheckedIndexedAccess in tsconfig while maintaining strong typing when accessing
+// values through indexes, ie: alphBalances[addressHash]
+export interface AddressesAlphBalances {
+  data: Record<AddressHash, DisplayBalances | undefined>
+  isLoading: boolean
+}
+
+const useAddressesAlphBalances = (addressHash?: AddressHash): AddressesAlphBalances => {
+  const { data: latestTxHashes, isLoading: isLoadingLatestTxHashes } = useAddressesLastTransactionHashes(addressHash)
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+
+  const { data, isLoading } = useQueries({
+    queries: latestTxHashes.map(({ addressHash, latestTxHash, previousTxHash }) =>
+      addressAlphBalancesQuery({ addressHash, latestTxHash, previousTxHash, networkId })
+    ),
+    combine
```

Comment:
What do you suggest to memoize? In the `useAddressesTokensBalances.ts` we need to merge the data returned from the `combine` with some other data, and for that it makes sense to `useMemo`. Here we don't have to do anything like that, we have what we need as the result of `combine`.

The data from `combine` will not change between re-renders of the same component unless the data returned from the query change. 

In `useAddressesTokensBalances.ts` we `useMemo` because we use `reduce` which returns always a new object. We don't want a new object if the deps haven't changed.

Reply:
Right! 🆗